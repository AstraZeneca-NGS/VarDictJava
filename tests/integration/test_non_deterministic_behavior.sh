#!/bin/bash -eu
set -o pipefail

# Load config vars
SCRIPT_DIR="$( cd "$( dirname "$0" )" && pwd )"
source $SCRIPT_DIR/config.sh

REF_GENOME="$DIR_REFERENCE/hg19.fa"
COLO20="$TESTS_DIR/non_deterministic_behavior/Colo829.chr20.PerlNonDeterm.bam"
REGION="-R chr20:61659400-61659600"

# Flag to remove differences because Perl doesn't sort unmapped reads
UNMAPPED_FLAG="-F 0x504"

N=5
for i in $(seq 1 $N); do 
	echo "Running VarDict Perl: $i / $N"
	time $VARDICTPERL \
			-G $REF_GENOME \
			-f 0.001 -N abc \
			-b $COLO20 $UNMAPPED_FLAG \
			-c 1 -S 2 -E 3 -g 4 \
			$REGION \
		| sort \
		> $DIR_OUTPUT/vardictColo20.perl.$i.txt

 	echo "Running VarDict Java: $i / $N"
 	time $VARDICTJAVA \
 			-G $REF_GENOME \
 			-f 0.001 -N abc \
 			-b $COLO20 $UNMAPPED_FLAG \
 			-c 1 -S 2 -E 3 -g 4 \
 			$REGION \
 	| sort \
 	> $DIR_OUTPUT/vardictColo20.java.$i.txt

done

echo "Differences:"
for i in $(seq 1 $N); do 
	for j in $(seq 1 $N); do 
		echo "--------------------------------------------------------------------------------"
		echo
		echo
		echo "Difference $i-$j Perl:"
		diff $DIR_OUTPUT/vardictColo20.perl.$i.txt $DIR_OUTPUT/vardictColo20.perl.$j.txt || true
		echo
		echo
		echo "Difference $i-$j Java:"
		diff $DIR_OUTPUT/vardictColo20.java.$i.txt $DIR_OUTPUT/vardictColo20.java.$j.txt || true
	done
done
