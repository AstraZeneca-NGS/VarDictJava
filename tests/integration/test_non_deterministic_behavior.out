Running VarDict Perl: 1 / 5
Running VarDict Java: 1 / 5
Running VarDict Perl: 2 / 5
Running VarDict Java: 2 / 5
Running VarDict Perl: 3 / 5
Running VarDict Java: 3 / 5
Running VarDict Perl: 4 / 5
Running VarDict Java: 4 / 5
Running VarDict Perl: 5 / 5
Running VarDict Java: 5 / 5
Differences:
--------------------------------------------------------------------------------


Difference 1-1 Perl:


Difference 1-1 Java:
--------------------------------------------------------------------------------


Difference 1-2 Perl:
8,9c8
< abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
< abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
---
> abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
11c10
< abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 1-2 Java:
--------------------------------------------------------------------------------


Difference 1-3 Perl:
11c11
< abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 1-3 Java:
--------------------------------------------------------------------------------


Difference 1-4 Perl:
8,9c8
< abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
< abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
---
> abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0


Difference 1-4 Java:
--------------------------------------------------------------------------------


Difference 1-5 Perl:
8,9c8
< abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
< abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
---
> abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
11c10
< abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 1-5 Java:
--------------------------------------------------------------------------------


Difference 2-1 Perl:
8c8,9
< abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
---
> abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
> abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
10c11
< abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 2-1 Java:
--------------------------------------------------------------------------------


Difference 2-2 Perl:


Difference 2-2 Java:
--------------------------------------------------------------------------------


Difference 2-3 Perl:
8c8,9
< abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
---
> abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
> abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0


Difference 2-3 Java:
--------------------------------------------------------------------------------


Difference 2-4 Perl:
10c10
< abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 2-4 Java:
--------------------------------------------------------------------------------


Difference 2-5 Perl:


Difference 2-5 Java:
--------------------------------------------------------------------------------


Difference 3-1 Perl:
11c11
< abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 3-1 Java:
--------------------------------------------------------------------------------


Difference 3-2 Perl:
8,9c8
< abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
< abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
---
> abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0


Difference 3-2 Java:
--------------------------------------------------------------------------------


Difference 3-3 Perl:


Difference 3-3 Java:
--------------------------------------------------------------------------------


Difference 3-4 Perl:
8,9c8
< abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
< abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
---
> abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
11c10
< abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 3-4 Java:
--------------------------------------------------------------------------------


Difference 3-5 Perl:
8,9c8
< abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
< abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
---
> abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0


Difference 3-5 Java:
--------------------------------------------------------------------------------


Difference 4-1 Perl:
8c8,9
< abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
---
> abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
> abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0


Difference 4-1 Java:
--------------------------------------------------------------------------------


Difference 4-2 Perl:
10c10
< abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 4-2 Java:
--------------------------------------------------------------------------------


Difference 4-3 Perl:
8c8,9
< abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
---
> abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
> abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
10c11
< abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 4-3 Java:
--------------------------------------------------------------------------------


Difference 4-4 Perl:


Difference 4-4 Java:
--------------------------------------------------------------------------------


Difference 4-5 Perl:
10c10
< abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 4-5 Java:
--------------------------------------------------------------------------------


Difference 5-1 Perl:
8c8,9
< abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
---
> abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
> abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0
10c11
< abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 5-1 Java:
--------------------------------------------------------------------------------


Difference 5-2 Perl:


Difference 5-2 Java:
--------------------------------------------------------------------------------


Difference 5-3 Perl:
8c8,9
< abc	chr20	chr20	61659444	61659444	A	G	23	17	4	0	15	2	A/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7778	0	0	2.000	2	4.4	14	18	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	0
---
> abc	chr20	chr20	61659444	61659444	A	AGAGGGGGCCGGTGGGGAGGGGGCCGGCGGGGAGTGGTTTGGTGGG	23	5	0	0	3	1	G/+45	0.2174	0;2	49.6	1	32.2	1	36.7	3.000	0.1579	0.1739	49	4.000	1	5.0	3	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	Insertion	0	5-0-0
> abc	chr20	chr20	61659444	61659444	A	G	23	17	0	0	15	2	G/G	0.7391	0;2	26.2	1	33.9	1	48.4	4.667	0.7368	0	0	2.000	2	4.4	14	19	CGGTGGGGAGGGGCCGGTGG	GAGGGGGCCGGTGGGGAGGG	chr20:61659400-61659600	SNV	0	5-0-0


Difference 5-3 Java:
--------------------------------------------------------------------------------


Difference 5-4 Perl:
10c10
< abc	chr20	chr20	61659471	61659483	CGGGGAGTGGTTT	TGGGGGGGGGCCTGCAGGGAGGGGGCC	27	8	14	3	5	3	GCGGGGAGTGGTTT/+28	0.2963	2;2	38.0	1	37.3	1	52.4	16.000	0.4000	0.2963	0	0	0	1.0	8	20	GCCGGTGGGGAGGGGGCCGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0
---
> abc	chr20	chr20	61659481	61659483	TTT	GCC	13	6	3	1	3	3	TTT/GCC	0.4615	2;2	35.0	1	37.3	1	60.0	12.000	0.8571	0.3846	0	3.000	1	0.8	6	7	AGGGGGCCGGCGGGGAGTGG	GGTGGGGAGGAGGCTGGTGA	chr20:61659400-61659600	Complex	0	0


Difference 5-4 Java:
--------------------------------------------------------------------------------


Difference 5-5 Perl:


Difference 5-5 Java:
